### PR TITLE
KOGITO-9144 Added information about how to enable service discovery

### PR DESCRIPTION
**Pull request by hbelmiro**
**JIRA:** https://issues.redhat.com/browse/KOGITO-9144

This PR depends on:

- https://github.com/kiegroup/kogito-runtimes/pull/2986

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] The nav.adoc file has a link to this guide in the proper category
- [x] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc [hunk 1]
```diff
@@ -168,35 +168,27 @@ Kubernetes service discovery is disabled::
...
-====
+[[ref-enabling-kubernetes-service-discovery]]
+== Enabling Kubernetes service discovery
+
 When using the Kubernetes service discovery feature, you need to balance if your application can afford the delayed startup time.
 
 If the URI pattern is not found in the application properties, then discovery is not triggered. However, the scanning is performed anyway. Therefore, a short time is required to go through the startup scan.
 
-You can disable the Kubernetes service discovery by removing the `kogito-addons-quarkus-kubernetes` library from the application's dependencies as shown in the following Maven exclusions:
+You can enable the Kubernetes service discovery by adding a service discovery implementation to the application's dependencies as shown in the following Maven dependencies:
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:178]: If I get it right, Service Discovery is deactivated by default, which means we need to specific the non backward into the release notes

Comment (hbelmiro) [reply serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:178]: Do you mean adding it to the release notes in this PR or the PR we usually have at the end of the release cycle?

Comment (radtriste) [reply serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:178]: To be added to the release notes once we create it for 1.39

**Review by kaldesai — APPROVED** [review]
@hbelmiro Your changes look good to me. Thank you!

**Review by fjtirado — APPROVED** [review]

**Review by MarianMacik — APPROVED** [review]

**Pull request merged**